src/python/pants/backend/project_info/tasks/export_dep_as_jar.py

```diff
@@ -299,9 +324,19 @@ def _full_library_set_for_target(target):
         libraries_for_target = set(
             [self._jar_id(jar) for jar in iter_transitive_jars(current_target)]
         )
-        for dep in sorted(flat_non_modulizable_deps_for_modulizable_targets[current_target]):
-            libraries_for_target.update(_full_library_set_for_target(dep))
-        info["libraries"].extend(libraries_for_target)
+        compile_libraries_for_target = libraries_for_target.copy()
+        for dep in sorted(
+            flat_non_modulizable_deps_for_modulizable_targets[current_target].compile_deps
+        ):
+            compile_libraries_for_target.update(_full_library_set_for_target(dep))
+        info["compile_libraries"].extend(compile_libraries_for_target)
+
+        runtime_libraries_for_target = libraries_for_target.copy()
+        for dep in sorted(
+            flat_non_modulizable_deps_for_modulizable_targets[current_target].runtime_deps
+        ):
+            runtime_libraries_for_target.update(_full_library_set_for_target(dep))
+        info["runtime_libraries"].extend(runtime_libraries_for_target)
```

Comment:
Given `compile_libraries_for_target` and `runtime_libraries_for_target` are `set`s, there should be no need to use `sort` with the for loop?

Reply:
You're right, since we used `FrozenOrderedSet`